File: 203-RemoveLinkedListElements/203-RemoveLinkedListElements.py
# Last updated: 09/07/2026, 10:09:11
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummy = ListNode(-1)
        dummy.next = head
        temp = dummy
        
        while temp and temp.next:
            if temp.next.val == val:
                temp.next = temp.next.next
 
            else:
                temp = temp.next
        return dummy.next

File: 203-RemoveLinkedListElements/203-RemoveLinkedListElements.java
// Last updated: 09/07/2026, 10:09:05
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode dummy = new ListNode(-1);
        ListNode tail = dummy;
        ListNode cur = head;
        while(cur!=null){
            if(cur.val!=val){
                tail.next = cur;
                tail = tail.next;
            }
            cur = cur.next;
        }
        tail.next = null;
        return dummy.next;
    }
}